src/mixins/atoms.js

```diff
@@ -88,13 +88,50 @@ export const AtomsMixin = (superclass) =>
         createAtomsGroup(basis, atomRadiiScale) {
             const atomsGroup = new THREE.Group();
             atomsGroup.name = ATOM_GROUP_NAME;
+            const { atomicLabelsArray, elementsWithLabelsArray } = basis;
             basis.coordinates.forEach((atomicCoordinate, atomicIndex) => {
                 const element = basis.getElementByIndex(atomicIndex);
                 const sphereMesh = this.getSphereMeshObject({
                     ...this._getDefaultSettingsForElement(element, atomRadiiScale),
                     coordinate: atomicCoordinate.value,
                 });
                 sphereMesh.name = `${element}-${atomicIndex}`;
+                sphereMesh.userData = {
+                    ...sphereMesh.userData,
```

Comment:
Did you just come up with this `userData` concept or was it used before?

Maybe `metadata` would be a better name - is there really data that comes from user here??

Reply:
No, this is THREE.js in-build field to store custom data <https://threejs.org/docs/#api/en/core/Object3D.userData>. If we create our own data fields (e.g., `metadata`), I think it will stripped while cloning THREE objects.